File: python/examples/quickstart_default_handler.py
"""Minimal agent: a single version with the smallest useful schema. Downloads an input
file, asks an LLM to summarise it, and uploads a result file.

Every agent should declare at least one schema — it defines the buyer-facing form and
the output view, so without one there's no published contract buyers can call. This is
the smallest useful schema; see versioned_schema_agent.py for progress reporting,
optional fields, and enums.

Run:
    export Z3T_AGENT_KEY=...
    python examples/quickstart_default_handler.py
"""

import asyncio
import os

from z3t_ai_agent import Agent, VersionSchema, s

agent = Agent(api_key=os.environ["Z3T_AGENT_KEY"])

summary_schema = VersionSchema(
    input=s.object({"document": s.file_uri(title="Document")}),
    output=s.object(
        {
            "summary": s.markdown(title="Summary"),
            "report": s.file_output(title="Summary file"),
        }
    ),
)


@agent.handle(version=1, schema=summary_schema)
async def handle(input: dict, ctx) -> dict:
    file = await ctx.files.download(input["document"])

    response = await ctx.llm.openai.chat.completions.create(
        model="gpt-4o",
        messages=[{"role": "user", "content": f"Summarise this document: {file.buffer[:2000]!r}"}],
    )
    summary = response.choices[0].message.content

    report_uri = await ctx.files.upload(summary.encode(), "summary.txt", "text/plain")

    return {
        "summary": summary,
        "report": report_uri,  # z3t://files/{id} — frontend renders a download button
    }


if __name__ == "__main__":
    asyncio.run(agent.start())

File: typescript/examples/quickstart-default-handler.ts
/**
 * Minimal agent: a single version with the smallest useful schema. Downloads an input
 * file, asks an LLM to summarise it, and uploads a result file.
 *
 * Every agent should declare at least one schema — it defines the buyer-facing form and
 * the output view, so without one there's no published contract buyers can call. This is
 * the smallest useful schema; see versioned-schema-agent.ts for progress reporting,
 * optional fields, and enums.
 *
 * Run:
 *   export Z3T_AGENT_KEY=...
 *   npx tsx examples/quickstart-default-handler.ts
 */

import { Agent, s } from '@z3t-ai/agent-sdk'

const agent = new Agent({ apiKey: process.env.Z3T_AGENT_KEY! })

agent.handle(1, {
  input: s.object({ document: s.fileUri({ title: 'Document' }) }),
  output: s.object({
    summary: s.markdown({ title: 'Summary' }),
    report:  s.fileOutput({ title: 'Summary file' }),
  }),
}, async (input, ctx) => {
  const { buffer } = await ctx.files.download(input.document)

  const response = await ctx.llm.openai.chat.completions.create({
    model: 'gpt-4o',
    messages: [{ role: 'user', content: `Summarise this document: ${buffer.subarray(0, 2000).toString()}` }],
  })
  const summary = response.choices[0].message.content ?? ''

  const reportUri = await ctx.files.upload(Buffer.from(summary), 'summary.txt', 'text/plain')

  return {
    summary,
    report: reportUri, // z3t://files/{id} — frontend renders a download button
  }
})

agent.start()
